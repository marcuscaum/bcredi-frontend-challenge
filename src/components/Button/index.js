import React from 'react';
import PropTypes from 'prop-types';

import Button from './index.styled';

const ButtonComponent = ({ type, icon, children, ...props }) => (
  <Button type={type} {...props}>
    <span>{icon && icon}</span>
    {children}
  </Button>
);

ButtonComponent.propTypes = {
  type: PropTypes.string,
  icon: PropTypes.object,
};

ButtonComponent.defaultProps = {
  icon: null,
  type: 'primary',
};

export default ButtonComponent;
