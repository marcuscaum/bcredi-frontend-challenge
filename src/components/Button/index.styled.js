import styled from 'styled-components';

const Button = styled.button`
  border: none;
  background-color: ${({ theme, type }) => theme[type]};
  padding: 14px 10px;
  line-height: 1.25;
  color: white;
  font-size: 16px;
  text-align: center;
  font-weight: bold;
  border-radius: 3px;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;

  &:disabled {
    background-color: #a6b9dc !important;
    cursor: not-allowed;
  }

  &:hover {
    background-color: #4565a2;
  }

  span {
    position: absolute;
    left: 30px;
  }
`;

export default Button;
