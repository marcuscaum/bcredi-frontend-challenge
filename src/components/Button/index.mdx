---
name: Button
---


import { Playground, PropsTable } from 'docz'
import Button from './index'

# Button

<PropsTable of={Button} />

## Basic usage

<Playground>
  <Button>Click me</Button>
</Playground>