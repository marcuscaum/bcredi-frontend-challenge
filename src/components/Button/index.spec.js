import React from 'react';
import { shallow } from 'enzyme';

import Button from '.';

const setup = (
  props = {
    type: 'primary',
  }
) => shallow(<Button {...props} />);

describe('[Component] Button', () => {
  it('should render the component', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
