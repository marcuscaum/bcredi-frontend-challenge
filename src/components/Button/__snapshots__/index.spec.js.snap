// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Button should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonComponent
    icon={null}
    type="primary"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <span />,
        undefined,
      ],
      "type": "primary",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <span />,
          undefined,
        ],
        "type": "primary",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": null,
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
